ambry-tools/src/main/java/com.github.ambry/tools/admin/ServerAdminTool.java

```diff
@@ -365,6 +367,33 @@ public static void main(String args[]) throws Exception {
   }
 
   /**
+   * Gets the {@link PartitionId} in the {@code clusterMap} whose string representation matches {@code partitionIdStr}.
+   * @param partitionIdStr the string representation of the partition required.
+   * @param clusterMap the {@link ClusterMap} to use to list and process {@link PartitionId}s.
+   * @return the {@link PartitionId} in the {@code clusterMap} whose string repr matches {@code partitionIdStr}.
+   * {@code null} if {@code partitionIdStr} is {@code null}.
+   * @throws IllegalArgumentException if there is no @link PartitionId} in the {@code clusterMap} whose string repr
+   * matches {@code partitionIdStr}.
+   */
+  public static PartitionId getPartitionIdFromStr(String partitionIdStr, ClusterMap clusterMap) {
+    if (partitionIdStr == null) {
+      return null;
+    }
+    PartitionId targetPartitionId = null;
+    List<? extends PartitionId> partitionIds = clusterMap.getAllPartitionIds();
+    for (PartitionId partitionId : partitionIds) {
+      if (partitionId.isEqual(partitionIdStr)) {
+        targetPartitionId = partitionId;
+        break;
+      }
```

Comment:
This looked a little funny to me ("why isn't he using .contains()?") but then I saw you're doing a comparison of partitionId to a string with isEquals so looks good